Lib/glyphsLib/glyphdata.py

```diff
@@ -31,69 +31,224 @@
+    meaningful production name can't be constructed or when the AGL
+    specification would be violated, get_glyph() will use the bare glyph
+    name then.
+
+    Note:
+    - Glyph name is the full name, e.g. "brevecomb_acutecomb.case".
+    - Base name is the base part, e.g. "brevecomb_acutecomb"
+    - Suffix is e.g. "case".
+    """
+
+    def is_unicode_u_value(name):
+        return name.startswith("u") and all(
+            part_char in "0123456789ABCDEF" for part_char in name[1:]
+        )
+
+    base_name, dot, suffix = glyph_name.partition(".")
+
+    # First, look up the full glyph name and base name in the AGLFN and in
+    # PRODUCTION_NAMES.
+    if (
+        glyph_name in agl.AGL2UV
+        or base_name in agl.AGL2UV
+        or glyph_name in (".notdef", ".null")
+    ):
+        return glyph_name
+
+    if glyph_name in data.PRODUCTION_NAMES:  # e.g. ain_alefMaksura-ar.fina -> uniFD13
+        return data.PRODUCTION_NAMES[glyph_name]
+    if base_name in data.PRODUCTION_NAMES:
+        final_production_name = data.PRODUCTION_NAMES[base_name] + dot + suffix
+        if len(final_production_name) > 31:
+            return None
+        return final_production_name
+
+    # If we aren't looking at a ligature and the name still hasn't been found,
+    # the glyph probably has no Unicode semantics, so return None.
+    if "_" not in base_name:
+        return None
+
+    # So we have a ligature that is not mapped in PRODUCTION_NAMES. Split it up and
+    # look up the individual parts.
+    base_name_parts = base_name.split("_")
+
+    # If all parts are in the AGLFN list, the glyph name is our production
+    # name already.
+    if all(part in agl.AGL2UV for part in base_name_parts):
+        if len(glyph_name) > 31:
+            return None
+        return glyph_name
+
+    _character_outside_BMP = False
+    production_names = []
+    for part in base_name_parts:
+        if part in agl.AGL2UV:
+            production_names.append(part)
+        else:
+            part_production_name = data.PRODUCTION_NAMES.get(part)
+            if part_production_name:
+                production_names.append(part_production_name)
+
+                # Note if there are any characters outside the Unicode BMP, e.g.
+                # "u10FFF" or "u10FFFF". Do not catch e.g. "u013B" though.
+                if len(part_production_name) > 5 and is_unicode_u_value(
+                    part_production_name
+                ):
+                    _character_outside_BMP = True
+
+            else:
+                return None
+
+    # Some names Glyphs uses resolve to other names that are not uniXXXX names and may
+    # contain dots (e.g. idotaccent -> i.loclTRK). If there is any name with a "." in
+    # it before the last element, punt. We'd have to introduce a "." into the ligature
+    # midway, which is invalid according to the AGL. Example: "a_i.loclTRK" is valid,
+    # but "a_i.loclTRK_a" isn't.
```

Comment:
that's correct according to AGLFN, I just hope that Glyphs.app's names also always conform to that rule.. Do they?

Reply:
Glyphs turns "a_idotaccent_a" into "a_i.loclTRK_a".

Reply:
that's not good. What does your code do in that case?

Reply:
https://github.com/googlei18n/glyphsLib/pull/387/files#diff-969a14671a2b5301f379a9ffe3bb13f6R56 😁

Reply:
I check the ‘a_i.loclTRK_a’ name. That is clearly wrong.